iceoryx_posh/test/moduletests/test_roudi_memory_manager.cpp

```diff
@@ -89,9 +119,7 @@ TEST_F(RouDiMemoryManager_Test, RouDiMemoryManagerDTorTriggersMemoryProviderDest
         RouDiMemoryManager sutDestroy;
         sutDestroy.addMemoryProvider(&memoryProvider1);
         sutDestroy.createAndAnnounceMemory();
-        EXPECT_CALL(memoryBlock1, destroyMock()).Times(1);
     }
-    EXPECT_CALL(memoryBlock1, destroyMock()).Times(0);
```

Comment:
same here. It's important to check that the destruction was triggered

Reply:
Thanks for the comment, i was wondering why was it useful ? :)

Reply:
The MemoryProvider tells the MemoryBlock that the memory will be destroyed, which means they must call the destructor of the object they store. These functions are pure virtual and needs to be implemented by the concrete memory block. We just need to check that the MemoryProvider works correctly and the chain of destruction is not broken. Unfortunately, we do not yet have enough documentation for this. I plan to add some when the SegmentManager is refactored.